### PR TITLE
DAOS-6071 csum: Checksum Scrubber Props

### PR DESCRIPTION
**Pull request by ryon-jensen**
Added new pool and container props as defined in data_integrity.md
for the checksum scrubbing feature.

Signed-off-by: Ryon Jensen <ryon.jensen@intel.com>

### CODE REVIEW

File: src/common/prop.c [hunk 1]
```diff
@@ -319,9 +319,24 @@ daos_prop_valid(daos_prop_t *prop, bool pool, bool input)
 				prop->dpp_entries[i].dpe_str))
 				return false;
 			break;
-		/* container-only properties */
 		case DAOS_PROP_PO_SVC_LIST:
 			break;
+		case DAOS_PROP_PO_SCRUB_SCHED:
+			val = prop->dpp_entries[i].dpe_val;
+			if (val != DAOS_SCRUB_SCHED_OFF &&
```
Comment (jolivier23) [inline src/common/prop.c:326]: If these values are contiguous, could we do something like >= and <= to cut down on checks (with some static asserts to ensure they remain so)

File: src/pool/rpc.c [hunk 2]
```diff
@@ -252,6 +252,15 @@ pool_query_bits(daos_pool_info_t *po_info, daos_prop_t *prop)
 		case DAOS_PROP_PO_OWNER_GROUP:
 			bits |= DAOS_PO_QUERY_PROP_OWNER_GROUP;
 			break;
+		case DAOS_PROP_PO_SCRUB_SCHED:
```
Comment (jolivier23) [inline src/pool/rpc.c:255]: also wonder if there is a better way to do this...though I guess it's probably not a critical path

**Review by jolivier23 — APPROVED** [review]

File: src/control/lib/control/pool_property.go [hunk 3]
```diff
@@ -116,6 +116,64 @@ func PoolProperties() PoolPropertyMap {
...
+					}
+
+					return &PoolPropertyValue{rsPct}, nil
+				},
+				valueStringer: func(v *PoolPropertyValue) string {
+					n, err := v.GetNumber()
+					if err != nil {
+						return "not set"
+					}
+					return fmt.Sprintf("%d%%", n)
```
Comment (mjmac) [inline src/control/lib/control/pool_property.go:172]: Are scrubbing credits really a percentage? Of what?

**Review by mjmac — CHANGES_REQUESTED** [review]
I think the property handlers in the control lib need a bit more work... Other stuff looks OK to me.

File: src/pool/rpc.c (hunk 2 above)
Comment (ryon-jensen) [reply src/pool/rpc.c:255]: Not sure what you mean? I just copied what is already being done. Are you thinking of reworking the whole prop query bits approach?

File: src/common/prop.c (hunk 1 above)
Comment (ryon-jensen) [reply src/common/prop.c:326]: I agree that's better. I'll look into doing that.

File: src/container/rpc.h [hunk 4]
```diff
@@ -217,22 +217,23 @@ CRT_RPC_DECLARE(cont_close, DAOS_ISEQ_CONT_CLOSE, DAOS_OSEQ_CONT_CLOSE)
...
-#define DAOS_CO_QUERY_PROP_OWNER_GROUP		(1ULL << 13)
-#define DAOS_CO_QUERY_PROP_DEDUP		(1ULL << 14)
-#define DAOS_CO_QUERY_PROP_DEDUP_THRESHOLD	(1ULL << 15)
-#define DAOS_CO_QUERY_PROP_ROOTS		(1ULL << 16)
-#define DAOS_CO_QUERY_PROP_CO_STATUS		(1ULL << 17)
-#define DAOS_CO_QUERY_PROP_ALLOCED_OID		(1ULL << 18)
-#define DAOS_CO_QUERY_PROP_EC_CELL_SZ		(1ULL << 19)
-
-#define DAOS_CO_QUERY_PROP_BITS_NR		(20)
+#define DAOS_CO_QUERY_PROP_SCRUB_DIS		(1ULL << 6)
```
[the remark this answers is not in the document]
Comment (ryon-jensen) [reply src/container/rpc.h:220]: That's a good point. I was just trying to keep the checksum related bits together, but am going to change it.

File: src/control/cmd/drpc_test/hello/drpc_test.pb.go [hunk 5]
```diff
@@ -7,7 +7,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 // 	protoc-gen-go v1.26.0
-// 	protoc        v3.6.1
+// 	protoc        v3.17.2
```
[the remark this answers is not in the document]
Comment (ryon-jensen) [reply src/control/cmd/drpc_test/hello/drpc_test.pb.go:10]: Makes sense and good to know. My first time updating proto stuff.

File: src/control/lib/control/pool_property.go [hunk 6]
```diff
@@ -116,6 +116,64 @@ func PoolProperties() PoolPropertyMap {
 				jsonNumeric: true,
 			},
 		},
+		"scrub": {
+			Property: PoolProperty{
+				Number:      drpc.PoolPropertyScrubSched,
+				Description: "Checksum scrubbing schedule",
```
[the remark this answers is not in the document]
Comment (ryon-jensen) [reply src/control/lib/control/pool_property.go:122]: Ya, policy is probably a better description. Will update

File: src/control/lib/control/pool_property.go (hunk 3 above)
Comment (ryon-jensen) [reply src/control/lib/control/pool_property.go:172]: No, neither is frequency. My first go code ... a lot was copy/pasted. Thanks for the feedback.

**Review by mjmac — APPROVED** [review]

**Pull request merged**